### PR TITLE
Fix tf2 compatibility of tf python implementation

### PR DESCRIPTION
**Pull request by roehling**
The Python code is completely replaced by forwarding calls to tf2_ros. This makes tf fully compatible with tf2, in particular it adds the missing support for static transformations.

A few API calls are no longer supported in the tf2 Python implementation. Most notably, the current pull request does not implement `chain()`, `frameExists()`, and `getFrameStrings()`. I believe those methods are rarely used and therefore expendable.

The `lookupTwist()` and `getLatestCommonTime()` implementations rely on ros/geometry2#192

### CODE REVIEW

**Comment by roehling** [comment]
Addresses #117

File: tf/package.xml [hunk 1]
```diff
@@ -41,11 +40,13 @@ any desired point in time.
...
   <run_depend>std_msgs</run_depend>
-  <run_depend version_gte="0.4.5">tf2</run_depend>
   <run_depend version_gte="0.4.5">tf2_ros</run_depend>
 
+  <test_depend>rostest</test_depend>
+  <test_depend>rosunit</test_depend>
+
   <export>
-    <roswtf plugin="tf.tfwtf" />
+    <!-- <roswtf plugin="tf.tfwtf" /> -->
```
Comment (tfoote) [inline tf/package.xml]: Does tfwtf not work anymore?

File: tf/test/testPython.py [hunk 2]
```diff
@@ -78,10 +83,10 @@ def elapsed_time_within_epsilon(t, delta, epsilon):
         epsilon = 0.1
 
         # Check for dedicated thread exception, existing frames
-        self.assertRaises(tf.Exception, lambda: t.waitForTransform("PARENT", "THISFRAME", rospy.Time(), timeout))
+        ###self.assertRaises(tf.Exception, lambda: t.waitForTransform("PARENT", "THISFRAME", rospy.Time(), timeout))
```
Comment (tfoote) [inline tf/test/testPython.py]: Why are these asserts commented out?

**Comment by tfoote** [comment]
@ros-pull-request-builder retest this please

File: tf/package.xml (hunk 1 above)
Comment (roehling) [reply tf/package.xml]: tfwtf is written for tf1 (e.g. it explicitly subscribes to `/tf`), so IMHO this functionality should be implemented by tf2 instead.

Comment (roehling) [reply tf/package.xml]: It should still work though, since the implementation details remain compatible enough for now.

**Comment by roehling** [comment]
Apparently, the failed check is caused by the unit tests of `tf_conversions`. A few functions have been marked deprecated, and the resulting compiler warnings make Jenkins declare the build unstable.

**Comment by roehling** [comment]
Is there anything else you need me to fix before this can get merged?

**Comment by tfoote** [comment]
Now that tf2 has been released with the upstream changes, it needs a version dependency. Other than that I've just been starved for maintenance time.

**Comment by roehling** [comment]
I took the liberty to fix the unstable build, even though it took me a bit longer than I care to admit ;-)

**Comment by tfoote** [comment]
Great, thanks for the contributions! I will shoot to merge and release this after we get the pending sync out.

**Comment by tfoote** [comment]
rebased and merged in #149

**Pull request closed without merging**